### PR TITLE
#745: Fixing external transformations

### LINKED ISSUE #730: Get transformations working with new model change
Sub-ticket of #607 
Follows on from #695 

The hardest part of this is going to be the depends on chain as there isn't any real hierarchy now or at least not one that we can traverse easily because each transform doesn't currently contain a reference to its parent. 

The model has been mostly implemented in `model/transformations.py` but will need to be modified to contain the `depends_on` chain logic

Also implement a suitable replacement for checking equality between two transforms.

### LINKED ISSUE #740: Rename Component.transform property
Rename to something more helpful, perhaps qtransform or final_3dview_transform.

Add return type hints for `transform` and `transforms`.

### LINKED ISSUE #745: Model change: Fix external transformation dependencies
The transform depends_on chain does not fully work yet as external transformations (ie having a depends_on as another transformation in a separate component) do not work.

### PR DESCRIPTION
**Pull request by rerpha**
### Issue

Closes #745 
Closes #740 
Closes #730 

### Description of work

- Fixes `depends_on` chain for local transformations 
- Allows external transformations by using a reference to the depended on component in `LinkTransformation`
- Removes horrible sample UI logic and treats that as a component, to simplify the 3d view. 

### Acceptance Criteria 

- Adding a link should work: add component1, add transformation, add component2, add link to component1, add transformation, should be dependent on the first component's position.
- Deleting a link should also work - follow steps above but delete link, second component should be moved to position relative to origin/sample rather than 

### UI tests

- Updated to use new model
- Some unit tests enabled and fixed

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

**Comment by rerpha** [comment]
still wip, currently all transforms are broken while i try and sort out the depends_on chain for everything including links and local transforms (local transforms are broken i think)

File: nexus_constructor/component_tree_model.py [hunk 1]
```diff
@@ -71,15 +71,15 @@ def _get_transformation_list(self, node, parent_item):
         transformation_list = None
         target_index = QModelIndex()
         if isinstance(parent_item, Component):
-            if not hasattr(parent_item, "stored_transforms"):
-                parent_item.stored_transforms = parent_item.transforms
-            transformation_list = parent_item.stored_transforms
+            # if not hasattr(parent_item, "stored_transforms"):
+            #     parent_item.stored_transforms = parent_item.transforms
```
Comment (matthew-d-jones) [inline nexus_constructor/component_tree_model.py:75]: remove?

File: nexus_constructor/instrument_view.py [hunk 2]
```diff
@@ -200,7 +200,10 @@ def add_component(
 
         mesh = OffMesh(geometry.off_geometry, self.component_root_entity, positions)
         material = create_material(
-            QColor("black"), QColor("grey"), self.component_root_entity
+            QColor("black") if name != "sample" else QColor("red"),
```
Comment (matthew-d-jones) [inline nexus_constructor/instrument_view.py:203]: Use `SAMPLE_NAME` from model.instrument

File: nexus_constructor/model/attribute.py [hunk 3]
```diff
@@ -13,7 +14,13 @@ class FieldAttribute:
     """
 
     name = attr.ib(type=str)
-    values = attr.ib(type=ValueType)
+    values = attr.ib(type=ValueType, cmp=False)
+
+    def __eq__(self, other_attribute):
```
Comment (matthew-d-jones) [inline nexus_constructor/model/attribute.py:19]: Would be good to cover with unit test

**Pull request merged**